Extensions/Firebase/C_firebasetools/D_cloudfirestoretools.js

```diff
@@ -0,0 +1,216 @@
+          callbackStateVariable.setString("ok");
+    })
+    .catch(function(error) {
+        if (callbackStateVariable)
+          callbackStateVariable.setString(error.message);
+    });
+}
+
+/**
+ * Gets a document and store it in a variable.
+ * @param {string} collectionName - The collection where the document is stored.
+ * @param {string} documentName - The name of the document to get.
+ * @param {gdjs.Variable} [callbackValueVariable] - The variable where to store the result.
+ * @param {gdjs.Variable} [callbackStateVariable] - The variable where to store if the operration was successful.
+ */
+gdjs.evtTools.firebase.firestore.getDocument = function(collectionName, documentName, callbackValueVariable, callbackStateVariable) {
+    firebase.firestore().collection(collectionName).doc(documentName).get()
+    .then(function(doc) {
+        if (callbackStateVariable)
+          callbackStateVariable.setString("ok");
+        gdjs.evtTools.network._objectToVariable(doc.data(), callbackValueVariable);
+    })
+    .catch(function(error) {
+        if (callbackStateVariable)
+          callbackStateVariable.setString(error.message);
+    });
+}
+
+/**
+ * Gets a field of a document and store it in a variable.
+ * @param {string} collectionName - The collection where the document is stored.
+ * @param {string} documentName - The name of the document.
+ * @param {string} field - The field to get.
+ * @param {gdjs.Variable} [callbackValueVariable] - The variable where to store the result.
+ * @param {gdjs.Variable} [callbackStateVariable] - The variable where to store if the operration was successful.
+ */
+gdjs.evtTools.firebase.firestore.getField = function(collectionName, documentName, field, callbackValueVariable, callbackStateVariable) {
+    firebase.firestore().collection(collectionName).doc(documentName).get()
+    .then(function(doc) {
+        if (callbackStateVariable)
+          callbackStateVariable.setString("ok");
+        gdjs.evtTools.network._objectToVariable(doc.get(field), callbackValueVariable);
+    })
+    .catch(function(error) {
+        if (callbackStateVariable)
+          callbackStateVariable.setString(error.message);
+    });
+}
+
+/**
+ * Checks for existence of a document.
+ * @param {string} collectionName - The collection where the document is stored.
+ * @param {string} documentName - The name of the document to check.
+ * @param {gdjs.Variable} [callbackValueVariable] - The variable where to store the result.
+ * @param {gdjs.Variable} [callbackStateVariable] - The variable where to store if the operration was successful.
+ */
+gdjs.evtTools.firebase.firestore.hasDocument = function(collectionName, documentName, callbackValueVariable, callbackStateVariable) {
+    firebase.firestore().collection(collectionName).doc(documentName).get()
+    .then(function(doc) {
+        if (callbackStateVariable)
+          callbackStateVariable.setString("ok");
+        callbackValueVariable.setNumber(doc.exists ? 1 : 2);
```

Comment:
1 and 2 are for passing a Boolean, as variables don't have those I use numbers: 1 for true and 2 for false. I do not use 0 because it's the default value of variables already. The best would be to be able to use a condition but it is not possible with those asynchronous interfaces.

Reply:
Personally I think numbers are pretty clear but I see how replacing them with a true and false string night be more intuitive.